contracts/state-validator/src/verifications/finalize_withdrawal.rs

```diff
@@ -0,0 +1,111 @@
+use core::result::Result;
+
+use gw_state::ckb_smt::smt::{Pair, Tree};
+use gw_utils::{
+    ckb_std::debug,
+    error::Error,
+    gw_common::H256,
+    gw_types::{
+        bytes::Bytes,
+        packed::{GlobalState, RawL2Block, RollupConfig, RollupFinalizeWithdrawalReader},
+        prelude::{Builder, Entity, Unpack},
+    },
+};
+
+pub mod last_finalized_withdrawal;
+mod user_withdrawal_cells;
+
+#[must_use]
+pub fn verify(
+    rollup_type_hash: &H256,
+    config: &RollupConfig,
+    args: RollupFinalizeWithdrawalReader,
+    prev_global_state: GlobalState,
+    post_global_state: GlobalState,
+) -> Result<(), Error> {
+    debug!("verify finalized withdrawal");
+
+    // Check global state version is 2
+    if 2 != prev_global_state.version_u8() || 2 != post_global_state.version_u8() {
```

Comment:
The global state is `2`, but the molecule state is `GlobalStateV1`, it confuses..